AsyncDisplayKit/Details/ASEnvironment.m

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "ASEnvironment.h"
+
+ASEnvironmentLayoutOptionsState ASEnvironmentLayoutOptionsStateCreate()
+{
+  return (ASEnvironmentLayoutOptionsState) {
+    .spacingBefore = 0,
```

Comment:
Do we need this default here, And if so, why not spacing after or others? It seems fine to include everything explicitly, but I am not sure that it is required either. Happy to go with whatever your intuition prefers.